src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/NewAzureStorageAccount.cs

```diff
@@ -134,6 +134,40 @@ public PSNetworkRuleSet NetworkRuleSet
             get; set;
         }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Enable Azure Files AAD Integration for the Storage account.")]
+        [ValidateNotNullOrEmpty]
+        public bool EnableAzureFilesAadIntegration
+        {
+            get
+            {
+                return enableAzureFilesAadIntegration.Value;
+            }
+            set
+            {
+                enableAzureFilesAadIntegration = value;
+            }
+        }
+        private bool? enableAzureFilesAadIntegration = null;
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Enable HierarchicalNamespace for the Storage account.")]
+        [ValidateNotNullOrEmpty]
+        public bool IsHnsEnabled
```

Comment:
instead of IsHnsEnabled, name I think the name should be "EnableHns", IsHnsEnabled makes sense when we are retrieving the properties, not setting them.
Also, change the type to a SwitchParameter instead of a bool. See [here](https://github.com/Azure/azure-powershell/blob/bbe5ee957de07032bcf7f9062bba91cceee62565/src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkResourceBase.cs#L32) for example

Reply:
As Mark discussed with me in the review of the design, it's good to be boo? fopr preview.